database/src/main/runs/define_flow.sql

```diff
@@ -0,0 +1,119 @@
+CREATE OR REPLACE FUNCTION runs.define_flow(
+    IN  i_segmentation      HSTORE,
+    IN  i_by_user           TEXT,
+    OUT status              INTEGER,
+    OUT status_text         TEXT
+) RETURNS record AS
+$$
+-------------------------------------------------------------------------------
+--
+-- Function: runs.define_flow(2)
+--      [Description]
+--
+-- Parameters:
+--      i_segmentation      - segmentation fo the flow
+--      i_by_user           - user initiating the linking
+--
+-- Returns:
+--      status             - Status code
+--      status_text        - Status text
+--
+-- Status codes:
+--      200     - OK
+--
+-------------------------------------------------------------------------------
+DECLARE
+    _segmentation_id    BIGINT;
+BEGIN
+    _segmentation_id = runs._get_key_segmentation(i_segmentation);
+
+    IF _segmentation_id IS NULL THEN
+        PERFORM 1
+        FROM runs._add_segmentation_flow(i_segmentation, i_by_user, TRUE, NULL);
+    ELSE
+    END IF;
+
+    -- TODO statuses & return values
+
+    RETURN;
+END;
+$$
+LANGUAGE plpgsql VOLATILE SECURITY DEFINER;
+
+GRANT EXECUTE ON FUNCTION runs.define_flow(HSTORE, TEXT) TO atum_user;
+
+CREATE OR REPLACE FUNCTION runs.define_flow(
+    IN  i_parent_segmentation   HSTORE,
+    IN  i_sub_segmentation      HSTORE,
+    IN  i_by_user               TEXT,
+    OUT status              INTEGER,
+    OUT status_text         TEXT
+) RETURNS record AS
+$$
+    -------------------------------------------------------------------------------
+--
+-- Function: runs.define_flow(3)
+--      [Description]
+--
+-- Parameters:
+--      i_key_user                - skypename of the user
+--
+-- Returns:
+--      status             - Status code
+--      status_text        - Status text
+--
+-- Status codes:
+--      200     - OK
+--
+-------------------------------------------------------------------------------
+DECLARE
+    _key_parent_segmentation        BIGINT;
+    _segmentation                   HSTORE;
+    _id_segmentation                BIGINT;
+    _take_pattern_additional_data   BOOLEAN;
+    _segmentation_id                BIGINT;
+BEGIN
+    _key_parent_segmentation := runs._get_key_segmentation(i_parent_segmentation);
+
+    IF _key_parent_segmentation IS NULL THEN
+        -- TODO error status parent does not exist
+        RETURN;
+    END IF;
+
+
+    _segmentation := i_parent_segmentation || i_sub_segmentation;
```

Comment:
Will check...

Reply:
OK, I looked in detail to the queries. Right now it's not handled at all - see the empty `ELSE - END IF` block on lines 110-111.
But perhaps it should add the flows of the parent segmentation? 🤔  WDYT?

Reply:
I'm not sure. I would assume existing flows should not be changed by new inserts. For me the question is, should the following segmentations all co-exist or are segments 1 and 2 mutually exclusive? I think they should co-exist

| id_seg | seg | key_parent_seg |
| --- | --- | --- |
| 1 | info_version => 1, info_date => 2022-03-23 | |
| 2 | info_version => 1, info_date => 2022-03-23 | 3 |
| 3 | info_date => 2022-03-23 | |

Reply:
Ok, so the semantics of 
1) `select runs.define_flow('info_date=>2022-03-23, info_version=>1'::hstore, 'kevin');`
vs
2) `select runs.define_flow('info_date=>2022-03-23'::hstore,'info_version=>1'::hstore, 'kevin');`

is basically create segment `info_date=>2022-03-23, info_version=>1` and in case 1) don't create a parent-child relationship, vs in case 2) create a parent-child relationship

Would you agree to that?